### PR TITLE
Configuration for Lint & Format CI Pipeline

### PR DESCRIPTION
**Pull request by candreso**
This PR will enable the repo's first GitHub Action, which will automatically lint and format our codebase 🥳! 
image

More specifically, this PR contains configurations and scripts to run eslint and prettier on both the server and client directories. (Each has different configs and slightly different scripts, which can be found in their respective package.json's.) This PR also contains the configuration for the GitHub Action workflow that will automatically run upon push and pull_request to main. 

**Note: In order to quickly get the linting pipeline to succeed on code that has already been committed to main, I ended up resolving eslint errors by simply disabling rules and commenting out the problematic code.** This means that someone needs to go back *actually* resolve the eslint errors at a later date to ensure that our code is up to standards.

This PR also contains other small edits to the codebase, which are detailed in the commit history.

### CODE REVIEW

**Comment by Xapollon430** [comment]
dope

File: .github/workflows/lint-format.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# author: candreso
+
+name: LintAndFormat
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ candreso-styling-config ]
+    # branches: [ main ]
```
Comment (candreso) [inline .github/workflows/lint-format.yml:14]: will change the branch when I push to main

File: .github/workflows/lint-format.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  pull_request:
+    branches: [ candreso-styling-config ]
+    # branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
```
Comment (candreso) [inline .github/workflows/lint-format.yml:19]: it looks like I still need to enable this

File: .github/workflows/lint-format.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "format"
+  format:
```
Comment (candreso) [inline .github/workflows/lint-format.yml:25]: I should also change this to lint-and-format

File: .github/workflows/lint-format.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+          
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "14.x"
+      - name: LintAndFormat
```
Comment (candreso) [inline .github/workflows/lint-format.yml:39]: I should also add documentation for this

File: client/src/components/Auth/AdminSignIn/AdminSignIn.js [hunk 5]
```diff
@@ -4,6 +4,7 @@ import FormikInit from "./AdminSignInSchema";
 import "../Sign.css";
 
 const AdminSignIn = () => {
+  // eslint-disable-next-line no-unused-vars
```
Comment (candreso) [inline client/src/components/Auth/AdminSignIn/AdminSignIn.js:7]: will try to actually resolve the eslint errors at a later date

File: client/src/components/Header/Header.js [hunk 6]
```diff
@@ -2,9 +2,8 @@ import styled from "styled-components";
 import "./Header.css";
 import React, { useState } from "react";
 import { Link, useHistory } from "react-router-dom";
-import { Button, Drawer, List } from "@material-ui/core";
```
Comment (candreso) [inline client/src/components/Header/Header.js:5]: I removed the unused List import for now, but I do want to use it (or something similar) to make the menu look nicer in the future

File: package.json [hunk 7]
```diff
@@ -7,7 +7,9 @@
     "server": "(cd server && npm run server)",
     "build-client": "(cd client && npm run build)",
     "win-install-all": "(cd client && npm install) && (cd ../server && npm install)",
-    "lin-install-all": "(cd client && npm install); (cd ../server && npm install)"
+    "lin-install-all": "(cd client && npm install); (cd ../server && npm install)",
+    "win-format": "(cd client && npm run format) && (cd ../server && npm run format)",
+    "lin-format": "(cd client && npm run format); (cd ../server && npm run format)"
```
Comment (candreso) [inline package.json:12]: the linux scripts still need to be tested

**Pull request merged**